doc/docs.md

```diff
@@ -2927,14 +2927,28 @@ The developer doesn't need to change anything in their code. "It just works", li
 Python, Go, or Java, except there's no heavy GC tracing everything or expensive RC for
 each object.
 
+### Control
+
+You can take advantage of V's autofree engine and define a `free()` method on custom 
+data types:
+
+```v
+fn (data MyType) free() {
```

Comment:
Here it should be: 
```v
[unsafe]
pub fn (data &MyType) free() {
```

Reply:
I am able to declare the below without `[unsafe]`:
```v
pub fn (data &MyType) free() {
    // ...
}
```
and call it without any warnings.

Reply:
Of course, calling it manually with -autofree enabled can result in invalid memory access (effectively called twice). I think it should produce a compiler error.

Reply:
`[unsafe]` is needed, so that calls to it will need to be in `unsafe{}` blocks.

This function is not safe to be called manually, because if one is not careful, it can lead to double frees, and leaves the struct in a partial state, i.e. it is intended to be called either by the -autofree mechanism, or manually by people, that should be aware of the risks, implementing their own functions marked with `[manualfree]`.

Functions marked by `[manualfree]` do not need to be marked as `[unsafe]` - they can provide an entirely safe interface to their callers. That tag simply means that the -autofree mechanism will not work inside them, and that the developer has the responsibility to free its own memory inside the marked function.

Reply:
> I am able to declare the below without `[unsafe]`:
> 
> ```v
> pub fn (data &MyType) free() {
>     // ...
> }
> ```
> 
> and call it without any warnings.

We may need to add a checker warning/error for .free methods that lack `[unsafe]` 🤔 .

Reply:
I would say we definitely need the warning(dev)/error(prod), if lacking `[unsafe]` or  `[manualfree]`.

Reply:
Created issue #9037.
Updated commit.